### PR TITLE
Support to filter classes based on modifiers

### PR DESCRIPTION
**Pull request by chia7712**
hi warbucks-maven-plugin
HBase plans to use this nice plugin to find out the classes having no specified annotations. However, the classes HBase cares are only the Public classes. This PR enable this plugin to filter classes based on the specified modifiers.

Related issue:
https://issues.apache.org/jira/browse/HBASE-19713

Best Regards
Ping

### CODE REVIEW

File: src/main/java/net/revelc/code/warbucks/maven/plugin/RuleProcessor.java [hunk 1]
```diff
@@ -94,6 +100,21 @@ long process() throws MojoExecutionException {
     };
   }
 
+  private Predicate<ClassInfo> isValidModifier() {
+    return clz -> {
+      int mod = clz.load().getModifiers();
```
[the remark this answers is not in the document]
Comment (chia7712) [reply src/main/java/net/revelc/code/warbucks/maven/plugin/RuleProcessor.java]: Thanks for the views. I will address your comments in next PR.

**Comment by chia7712** [comment]
I have added the test and doc. Please take a look. Thanks.

**Comment by ctubbsii** [comment]
Thanks for the contribution, @chia7712 . I'll merge this in and cut a release tomorrow, after a small bit of manual testing. Hopefully, these changes will help HBase use it for their use case.

**Comment by chia7712** [comment]
Any updates? It is a pleasure to use your plugin in hbase 2.0~

**Comment by ctubbsii** [comment]
Sorry for the delay. I got distracted by other things. I will merge in the next few minutes.

**Pull request merged**